File: app/src/main/java/ua/te/hackathon/smartcity2015/ui/BaseActivity.java
package ua.te.hackathon.smartcity2015.ui;

import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.google.android.gms.auth.api.Auth;
import com.google.android.gms.auth.api.signin.GoogleSignInResult;
import com.google.android.gms.common.api.OptionalPendingResult;

import java.util.Locale;

import ua.te.hackathon.smartcity2015.ui.intro.IntroActivity;

public abstract class BaseActivity extends AppCompatActivity {

  @Override
  public final void onCreate(Bundle savedInstanceState) {
    onPreCreate(savedInstanceState);
    super.onCreate(savedInstanceState);

    if (isUserAuthenticated()) {
      onCreateAuthenticated(savedInstanceState);
    } else {
      onCreateUnauthenticated(savedInstanceState);
    }
  }

  protected abstract void onCreateAuthenticated(Bundle savedInstanceState);

  protected void onPreCreate(Bundle savedInstanceState) {
    if (shouldLockOrientation()) {
      setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
    }

    setUkrainianLocale();
  }

  protected boolean shouldLockOrientation() {
    return true;
  }

  private boolean isUserAuthenticated() {
      return  true;

  }

  private void setUkrainianLocale() {
    Locale locale = new Locale("uk");
    Locale.setDefault(locale);
    Configuration config = new Configuration();
    config.locale = locale;
    getBaseContext().getResources().updateConfiguration(config,
        getBaseContext().getResources().getDisplayMetrics());
  }

  public void onCreateUnauthenticated(Bundle savedInstanceState) {
    Intent intent = IntroActivity.startRegistration(this);
    startActivity(intent);
    finish();
  }
}

File: app/src/main/kotlin/ua/te/hackathon/smartcity2015/ui/BaseActivity.kt
package ua.te.hackathon.smartcity2015.ui

import android.content.pm.ActivityInfo
import android.content.res.Configuration
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import ua.te.hackathon.smartcity2015.ui.intro.IntroActivity
import ua.te.hackathon.smartcity2015.utils.LoggedUser
import java.util.*

abstract class BaseActivity : AppCompatActivity() {

  public override fun onCreate(savedInstanceState: Bundle?) {
    onPreCreate(savedInstanceState)
    super.onCreate(savedInstanceState)

    if (isUserAuthenticated) {
      onCreateAuthenticated(savedInstanceState)
    } else {
      onCreateUnauthenticated(savedInstanceState)
    }
  }

  protected abstract fun onCreateAuthenticated(savedInstanceState: Bundle?)

  protected fun onPreCreate(savedInstanceState: Bundle?) {
    if (shouldLockOrientation()) {
      requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
    }

    setUkrainianLocale()
  }

  protected fun shouldLockOrientation(): Boolean {
    return true
  }

  private val isUserAuthenticated: Boolean
    get() = LoggedUser.isLoggedIn

  private fun setUkrainianLocale() {
    val locale = Locale("uk")
    Locale.setDefault(locale)
    val config = Configuration()
    config.locale = locale
    baseContext.resources.updateConfiguration(config,
        baseContext.resources.displayMetrics)
  }

  fun onCreateUnauthenticated(savedInstanceState: Bundle?) {
    val intent = IntroActivity.startRegistration(this)
    startActivity(intent)
    finish()
  }
}
